src/tests/user/profile.test.ts

```diff
@@ -0,0 +1,95 @@
+import request from 'supertest';
+import { faker } from '@faker-js/faker';
+import jwt from 'jsonwebtoken';
+
+import app from '../../index.js';
+import { AppDataSource } from '../../config/database.js';
+import { User } from '../../entities/User.js';
+import { configs } from '../../config/env.js';
+
+let randomUser: User;
+let validToken: string;
+let expiredToken: string;
+let invalidToken: string;
+let validTokenForNotRegisteredUser: string
+const getUserProfileURL = '/api/users/profile';
+
+beforeAll(async () => {
+  await AppDataSource.initialize();
+
+  const newUser = {
+    email: faker.internet.email(),
+    password: faker.internet.password(),
+    name: faker.person.fullName(),
+  };
+  const savedUser = await AppDataSource.getRepository(User).save(newUser);
+  randomUser = savedUser;
+
+  if (!configs.auth.JWT_SECRET) {
+    throw new Error('Error in verifing the token');
+  }
+  
+  validToken = jwt.sign({ id: randomUser.id }, configs.auth.JWT_SECRET, {
+    expiresIn: '1h',
+  });
+  validTokenForNotRegisteredUser = jwt.sign({ id: 100 }, configs.auth.JWT_SECRET, {
+    expiresIn: '1h',
+  });
+  invalidToken = jwt.sign({ id: randomUser.id }, 'invalidJWTSecret', {
+    expiresIn: '1h',
+  });
+  expiredToken = jwt.sign({ id: randomUser.id }, configs.auth.JWT_SECRET, {
+    expiresIn: '-1s',
+  });
+});
+
+afterAll(async () => {
+  await AppDataSource.getRepository(User).delete({});
```

Comment:
Good job here, If you replace user creation with register route, you could delete it by email.